app/models/post.rb

```diff
@@ -33,6 +31,16 @@ def create_vote
     validates :body, length: { minimum: 20 }, presence: true
     validates :topic, presence: true
     validates :user, presence: true
+    def create_vote
+      user.votes.create(value: 1, post: self)
+    end
+   
+    def save_with_initial_vote
+     ActiveRecord::Base.transaction do
+      self.save
+      self.create_vote
```

Comment:
is the `self` necessary here?  If you remove it, it will default to calling `save` and `create_vote` on the same class its currently in.